staNMF/staNMF.py

```diff
@@ -371,10 +392,16 @@ def instability(self, k1=0, k2=0):
                 self.instabilitydict[k] = (np.sum(distMat) / (numReplicates *
                                            (numReplicates-1)))
 
-                outputfile = open("instability.csv", "w")
-                for i in sorted(self.instabilitydict):
-                    outputfile.write(str(i) + "," +
-                                     str(self.instabilitydict[i]) + "\n")
+                if self.parallel:
```

Comment:
I don't think you'll get any sort of performance drop by initializing a `csv.writer()`. It is not worth avoiding though since there are [differences in new line characters depending on OS](http://stackoverflow.com/questions/1761051/difference-between-n-and-r)